lib/src/model.dart

```diff
@@ -1037,6 +1037,37 @@ class Report {
 
   static Report fromJson(Map<String, dynamic> json) => _$ReportFromJson(json);
   Map<String, dynamic> toJson() => _$ReportToJson(this);
+
+  /// Creates a new [Report] instance with [section] extending and already
+  /// existing [ReportSection]. The sections are matched via the `title`.
+  ///
+  /// The granted and max points will be added to the existing section.
+  /// The summary will be appended to the end of the existing summary.
+  ///
+  /// If there is no section matched, the section will be added to the end of
+  /// the sections list.
+  Report joinSection(ReportSection section) {
+    final matched = sections.firstWhere((s) => s.title == section.title,
+        orElse: () => null);
+    if (matched == null) {
+      return Report(sections: [...sections, section]);
+    } else {
+      return Report(
+          sections: sections.map(
+        (s) {
+          if (s != matched) {
+            return s;
+          }
+          return ReportSection(
+            title: s.title,
+            maxPoints: s.maxPoints + section.maxPoints,
+            grantedPoints: s.grantedPoints + section.grantedPoints,
+            summary: [s.summary.trim(), section.summary.trim()].join('\n\n'),
```

Comment:
Maybe we need to special case this a bit more.

The first paragraph of the section should explain the scoring rule of that section, followed by details of any issues.

Reply:
This points towards having a specific data structure for a section's generic explanations of the scoring rule, and a separate part for listing the issues. I'd rather start with this simpler concatenation, and as we evaluate the result, we can go back and restructure it?

Reply:
SG.

I don't think we had anticipated the combination of sections, and I think with sandboxing that won't be needed.